app/services/account_reset/delete_account.rb

```diff
@@ -73,6 +87,8 @@ def extra_analytics_attributes
         account_age_in_days: account_age,
         account_confirmed_at: user.confirmed_at,
         mfa_method_counts: mfa_method_counts,
+        identity_verified: verified_status,
+        identity_verification_method: verification_method,
```

Comment:
Similar comment to what I made in another pull request at https://github.com/18F/identity-idp/pull/11138#discussion_r1732725982 : The way we're assigning these as side effects seems rather fragile and requires a very specific order of operations to work the way we expect, when it feels like it'd work just as well to compute these on demand? 

i.e. something like...

```rb
def verified_status
  user.identity_verified?
end

def verification_method
  if user.pending_in_person_enrollment.present?
    :in_person_proofing
  elsif user.gpo_verification_pending_profile.present?
    :verify_by_mail
  elsif user.identity_verified_with_biometric_comparison?
    :biometric_comparison
  end
end
```

For the first one, it's simple enough that we could probably inline it.

```suggestion
        identity_verified: user.identity_verified?,
        identity_verification_method: verification_method,
```

Reply:
I think I see your meaning. By way of being run during the call action rather than just begin added on demand - instead should be assigned directly, right?

Reply:
I think it eliminates a lot of uncertainty to not have to worry about something having been assigned earlier in the code lifecycle. There might be some edge-case conditions where it's not always assigned the way we expect, or something that happens later that causes those values to be reset unexpectedly. I think it's also a mental disconnect of having to juggle both the assignment and the reading of that assignment. Particularly with assignment, would its purpose be clear to someone unfamiliar with the code? There's also a risk that if we remove these properties from analytics attributes, we're continuing to do wasted work to assign the values.

Rather, if we can just figure out what the values ought to be when we want them, it's all a bit simpler and self-contained.